morse_decoder.rb

```diff
@@ -0,0 +1,49 @@
+@morse_hash = {
+  '.-' => 'a',
+  '-...' => 'b',
+  '-.-.' => 'c',
+  '-..' => 'd',
+  '.' => 'e',
+  '..-.' => 'f',
+  '--.' => 'g',
+  '....' => 'h',
+  '..' => 'i',
+  '.---' => 'j',
+  '-.-' => 'k',
+  '.-..' => 'l',
+  '--' => 'm',
+  '-.' => 'n',
+  '---' => 'o',
+  '.--.' => 'p',
+  '--.-' => 'q',
+  '.-.' => 'r',
+  '...' => 's',
+  '-' => 't',
+  '..-' => 'u',
+  '...-' => 'v',
+  '.--' => 'w',
+  '-..-' => 'x',
+  '-.--' => 'y',
+  '--..' => 'z'
+}
+
+def decode_letter(letter)
+  current_letter = @morse_hash[letter]
+  current_letter.upcase
+end
+
+def decode_word(word)
+  letters_array = word.split
+  decoded_letters = letters_array.map { |letter| decode_letter(letter) }
+  decoded_word = decoded_letters.reduce { |accum, letter| accum + letter }
+  decoded_word.upcase
+end
+
+def decode_message(message)
+  word_array = message.split('   ')
+  decode_message = word_array.map { |word| decode_word(word) }
+  decode_phrase = decode_message.reduce { |accum, word| `#{accum} #{word}` }
```

Comment:
- [ ] There is a bug 🐛 when decoding the message and the message is not decoded completely as shown below 👇. Please try to fix 👍 
![image](https://user-images.githubusercontent.com/46839548/209236170-db4e11f7-7e5c-4b18-a752-09ae81f4c86c.png)